src/lib.rs

```diff
@@ -78,45 +83,82 @@ pub(crate) fn build_vector<Vector: VelloVector + Component>() -> SystemConfigs {
 #[allow(clippy::type_complexity)]
 fn build_fill_only_vector<Vector: VelloVector + Component>(
     mut q_vectors: Query<
-        (&Vector, &Fill, &mut VelloScene),
-        (Without<Stroke>, Or<(Changed<Vector>, Changed<Fill>)>),
+        (&Vector, &Fill, Option<&mut VelloArrow>, &mut VelloScene),
+        (
+            Without<Stroke>,
+            Or<(Changed<Vector>, Changed<Fill>, Changed<VelloArrow>)>,
+        ),
     >,
 ) {
-    for (vector, fill, mut scene) in q_vectors.iter_mut() {
+    for (vector, fill, arrow, mut scene) in q_vectors.iter_mut() {
         *scene = VelloScene::default();
 
         // Build the vector to the VelloScene
         vector.build_fill(fill, &mut scene);
+
+        // Build the possible arrow
+        if let Some(mut arrow) = arrow {
+            arrow.attach(vector.shape());
```

Comment:
I dont think its good to automatically `attach()` here, i think the user should be responsible for calling it before hand.